### PR TITLE
[libc] Make str_to_float independent of fenv

### PR DESCRIPTION
**Pull request by michaelrj-google**
The str_to_float conversion code doesn't need the features provided by
fenv and the dependency is creating a blocker for hand-in-hand. This
patch uses a workaround to remove this dependency.

### CODE REVIEW

File: libc/src/__support/str_to_float.h [hunk 1]
```diff
@@ -525,10 +525,13 @@ clinger_fast_path(ExpandedFloat<T> init_num,
...
-    float_mantissa = static_cast<T>(fputil::DyadicFloat<128>(
-        Sign::POS, 0,
-        fputil::DyadicFloat<128>::MantissaType(
-            {uint64_t(mantissa), uint64_t(mantissa >> 64)})));
+    // float_mantissa = static_cast<T>(fputil::DyadicFloat<128>(
+    //     Sign::POS, 0,
+    //     fputil::DyadicFloat<128>::MantissaType(
+    //         {uint64_t(mantissa), uint64_t(mantissa >> 64)})));
+    float_mantissa = (static_cast<T>(uint64_t(mantissa)) *
+                      static_cast<T>(1ull << 32) * static_cast<T>(1ull << 32)) +
```
Comment (lntue) [inline libc/src/__support/str_to_float.h:533]: you can use `static_cast<T>(0x1.0p64)` for 2^64.

**Review by lntue — APPROVED** [review]
LGTM with 1 nit.

File: libc/src/__support/str_to_float.h (hunk 1 above)
Comment (michaelrj-google) [reply libc/src/__support/str_to_float.h:533]: That's much cleaner, thank you.

**Review by lntue — APPROVED** [review]

**Comment by mikhailramalho** [comment]
@michaelrj-google this PR broke rv32 compilation: https://lab.llvm.org/staging/#/builders/196/builds/471/steps/8/logs/stdio

**Comment by michaelrj-google** [comment]
I'm looking into the rv32 issue

**Pull request merged**